tests/generic/core/test_SerialConsole.py

```diff
@@ -314,3 +317,23 @@ def test_SerialConsole_login_except_on_wrong_username_match(serial_console_proxy
 
 def test_SerialConsole_does_not_require_login(serial_console_proxy):
     assert serial_console_proxy.console.requires_login is True
+
+
+def test_SerialConsole_raw_log_file(pty_pair):
+    received = 'abcd\nefgh'
+
+    with tempfile.NamedTemporaryFile(delete=False) as tmpfile:
+        console = SerialConsole(
+            port=os.ttyname(pty_pair.main.fd),
+            baud=115200,  # Baud Doesn't really matter as virtual tty,
+            encoding='utf-8', raw_logfile=tmpfile.name)
+        # tmpfile.name
+
+        pty_pair.secondary.write(received)
+        time.sleep(1)
+        print(console.read_all())
+        time.sleep(1)
+        console.close()
+        time.sleep(1)
+
+        assert tmpfile.read() == received
```

Comment:
decode `bytes` to `str`.
```suggestion
        assert tmpfile.read().decode() == received
```